torchgeo/datamodules/levircd.py

```diff
@@ -1,150 +1,174 @@
+        if dataset_variant == 'levircd':
+            dataset_class = cast(type[LEVIRCD | LEVIRCDPlus], LEVIRCD)
+            # Default to None for official splits
+            if val_split_pct is None:
+                val_split_pct = None
+        elif dataset_variant == 'levircd-plus':
+            dataset_class = cast(type[LEVIRCD | LEVIRCDPlus], LEVIRCDPlus)
+            # Default to 0.2 for LEVIR-CD+ random splits
+            if val_split_pct is None:
+                val_split_pct = 0.2
+        else:
+            raise ValueError(
+                f"dataset_variant must be 'levircd' or 'levircd-plus', got {dataset_variant}"
+            )
+
         super().__init__(
-            LEVIRCDPlus, batch_size=batch_size, num_workers=num_workers, **kwargs
+            dataset_class, batch_size=batch_size, num_workers=num_workers, **kwargs
         )
 
         self.patch_size = _to_tuple(patch_size)
+        self.stride = _to_tuple(stride) if stride is not None else None
+        self.dataset_variant = dataset_variant
         self.val_split_pct = val_split_pct
 
-        self.train_aug = K.AugmentationSequential(
-            K.VideoSequential(
-                K.Normalize(mean=self.mean, std=self.std),
-                K.RandomCrop(self.patch_size, pad_if_needed=True),
-            ),
+        # Create transforms using factory methods
+        self.train_aug = self._create_random_train_aug()
+        self.val_aug = self._create_deterministic_val_aug()
+        self.test_aug = self._create_deterministic_val_aug()  # Same as val
+        self.aug = self.val_aug  # Fallback general augmentation
+
+    def _create_random_train_aug(self) -> K.AugmentationSequential:
+        """Create random training augmentation for maximum data diversity."""
+        return K.AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            _RandomNCrop(size=self.patch_size, pad_if_needed=True),
             data_keys=None,
             keepdim=True,
+            same_on_batch=False,  # Different crops per batch item
         )
-        self.val_aug = K.AugmentationSequential(
-            K.VideoSequential(
-                K.Normalize(mean=self.mean, std=self.std),
-                _ExtractPatches(window_size=self.patch_size),
-            ),
-            data_keys=None,
-            keepdim=True,
-            same_on_batch=True,
-        )
-        self.test_aug = K.AugmentationSequential(
-            K.VideoSequential(
-                K.Normalize(mean=self.mean, std=self.std),
-                _ExtractPatches(window_size=self.patch_size),
+
+    def _create_deterministic_val_aug(self) -> K.AugmentationSequential:
+        """Create deterministic validation/test augmentation for reproducible evaluation."""
+        return K.AugmentationSequential(
+            K.Normalize(mean=self.mean, std=self.std),
+            _ExtractPatches(
+                window_size=self.patch_size, stride=self.stride, keepdim=False
```

Comment:
The updated levir and transforms don't quite work for me. I think that this should be set to `keepdim=True`, so that the correct reshaping triggers.